app/webpacker/components/ContactEditProfilePage/EditProfileForm.jsx

```diff
@@ -0,0 +1,185 @@
+import React, { useEffect, useState } from 'react';
+import { Form, Message } from 'semantic-ui-react';
+import ReCAPTCHA from 'react-google-recaptcha';
+import { QueryClient, useQuery } from '@tanstack/react-query';
+import i18n from '../../lib/i18n';
+import { IdWcaSearch } from '../SearchWidget/WcaSearch';
+import SEARCH_MODELS from '../SearchWidget/SearchModel';
+import { apiV0Urls, contactEditProfileActionUrl } from '../../lib/requests/routes.js.erb';
+import { genders, countries } from '../../lib/wca-data.js.erb';
+import Loading from '../Requests/Loading';
+import Errored from '../Requests/Errored';
+import useSaveAction from '../../lib/hooks/useSaveAction';
+import { fetchJsonOrError } from '../../lib/requests/fetchWithAuthenticityToken';
+import UtcDatePicker from '../wca/UtcDatePicker';
+
+const CONTACT_EDIT_PROFILE_FORM_QUERY_CLIENT = new QueryClient();
+
+const genderOptions = _.map(genders.byId, (gender) => ({
+  key: gender.id,
+  text: gender.name,
+  value: gender.id,
+}));
+
+const countryOptions = _.map(countries.byIso2, (country) => ({
+  key: country.iso2,
+  text: country.name,
+  value: country.iso2,
+}));
+
+export default function EditProfileForm({
+  wcaId: loggedInUserWcaId,
+  loggedInUserData,
+  onContactSuccess,
+  recaptchaPublicKey,
+}) {
+  const [contactEmail, setContactEmail] = useState();
+  const [customWcaId, setCustomWcaId] = useState();
+  const [editProfileReason, setEditProfileReason] = useState();
+  const [editedProfileDetails, setEditedProfileDetails] = useState();
+  const [proof, setProof] = useState();
+  const [captchaValue, setCaptchaValue] = useState();
+  const [captchaError, setCaptchaError] = useState(false);
+  const [saveError, setSaveError] = useState();
+  const { save, saving } = useSaveAction();
+  const wcaId = loggedInUserWcaId || customWcaId;
+  const { data, isLoading, isError } = useQuery({
+    queryKey: ['profileData'],
+    queryFn: () => fetchJsonOrError(apiV0Urls.persons.show(wcaId)),
+  }, CONTACT_EDIT_PROFILE_FORM_QUERY_CLIENT);
+  const profileDetails = data?.data?.person;
```

Comment:
More line breaks for readability please :face_with_spiral_eyes:

Reply:
Done